modules/messages/client/components/Thread.component.js

```diff
@@ -0,0 +1,276 @@
+        <br />
+        <a href="/profile/edit">{t('Edit your profile')}</a>
+      </p>
+    </div>
+  );
+}
+
+function UserDoesNotExist() {
+  const { t } = useTranslation('messages');
+  return (
+    <div className="content-empty">
+      <i className="icon-3x icon-messages-alt" />
+      <h4>{t("This user isn't a member anymore.")}</h4>
+    </div>
+  );
+}
+
+function Loading() {
+  return (
+    <LoadingContainer>
+      <LoadingIndicator />
+    </LoadingContainer>
+  );
+}
+
+export default function Thread({ user, profileMinimumLength }) {
+  if (!user.public) {
+    return (
+      <section className="container-spacer">
+        <Activate />
+      </section>
+    );
+  }
+
+  const username = getRouteParams().username;
+
+  if (user.username === username) {
+    go('inbox');
+    return null; // important to return null to indicate "nothing to render"
+  }
+
+  const [nextParams, setNextParams] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [isFetchingMore, setIsFetchingMore] = useState(false);
+  const [otherUser, setOtherUser] = useState(null);
+  const [doesNotExist, setDoesNotExist] = useState(false);
+  const [messages, setMessages] = useState([]);
+  const cacheKey = `messages.thread.${user._id}-${username}`;
+
+  const hasEmptyProfile = useMemo(
+    () => plainTextLength(user.description) < profileMinimumLength,
+    [user],
+  );
+
+  const userHasReplied = Boolean(
+    messages.find(message => message.userFrom._id === user._id),
+  );
+  const showQuickReply = messages.length > 0 && !userHasReplied;
+  const showReply = !hasEmptyProfile;
+
+  const isExtraSmall = useMediaQuery({ maxWidth: 768 - 1 });
+
+  async function fetchMoreData() {
+    if (isFetchingMore || !nextParams) return;
+    setIsFetchingMore(true);
+    try {
+      const {
+        messages: moreMessages,
+        nextParams: moreNextParams,
+      } = await api.messages.fetchMessages(otherUser._id, nextParams);
+      setMessages(messages => [
+        ...moreMessages.sort((a, b) => a.created.localeCompare(b.created)),
+        ...messages,
+      ]);
+      setNextParams(moreNextParams);
+    } finally {
+      setIsFetchingMore(false);
+    }
+  }
+
+  async function fetchData() {
+    if (isFetching) return;
+    const username = getRouteParams().username;
+    try {
+      setIsFetching(true);
+      const otherUser = await api.users.fetch(username);
+      const { messages, nextParams } = await api.messages.fetchMessages(
+        otherUser._id,
+      );
+      setOtherUser(otherUser);
+      setMessages(messages.sort((a, b) => a.created.localeCompare(b.created)));
+      setNextParams(nextParams);
+    } catch (error) {
+      if (error.response?.status === 404) {
+        setDoesNotExist(true);
+      } else {
+        throw error;
+      }
+    } finally {
+      setIsFetching(false);
+    }
+  }
+
+  async function sendMessage(content) {
```

Comment:
Not that it's like this in `master` either, but would be great if focus would return to text input after sending the message. :-)

Reply:
Implemented it in https://github.com/Trustroots/trustroots/pull/1173/commits/1d43720c0cb78b4556673405d573d1546027fd12